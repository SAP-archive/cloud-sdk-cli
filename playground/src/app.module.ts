import { Module } from '@nestjs/common';
import { AppController } from './app.controller';
import { AppService } from './app.service';

import { CatalogueModule } from './catalogue/catalogue.module';

@Module({
  imports: [CatalogueModule],
  controllers: [AppController],
  providers: [AppService],
})
export class AppModule {}
