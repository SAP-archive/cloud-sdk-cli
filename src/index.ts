/*!
 * Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved.
 */

export { run } from '@oclif/command';
