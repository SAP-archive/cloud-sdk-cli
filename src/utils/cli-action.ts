/*!
 * Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved.
 */
import cli from 'cli-ux';

export async function action(output: string, showSpinner: boolean, promise: Promise<any>) {
  if (showSpinner) {
    cli.action.start(output);
  } else {
    cli.log(`${output}...`);
  }
  await promise;
  if (showSpinner) {
    cli.action.stop();
  }
}
