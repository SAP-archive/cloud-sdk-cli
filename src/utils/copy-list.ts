/*!
 * Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved.
 */

export interface CopyDescriptor {
  sourcePath: string | URL;
  targetFolder: string;
  fileName: string;
}
