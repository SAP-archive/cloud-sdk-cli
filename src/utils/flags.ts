/*!
 * Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved.
 */

import Command from '@oclif/command';
import { OutputFlags } from '@oclif/parser';

type FlagsInput<CommandT extends typeof Command> = CommandT['flags'];

export type Flags<CommandT extends typeof Command> = OutputFlags<NonNullable<FlagsInput<CommandT>>>;
export type RequiredFlagsParam<CommandT extends typeof Command, ParamsT extends keyof Flags<CommandT>> = Pick<Flags<CommandT>, ParamsT>;
export type OptionalFlagsParam<CommandT extends typeof Command, ParamsT extends keyof Flags<CommandT>> = Partial<
  RequiredFlagsParam<CommandT, ParamsT>
>;
export type FlagsParam<
  CommandT extends typeof Command,
  RequiredParamsT extends keyof Flags<CommandT>,
  OptionalParamsT extends keyof Flags<CommandT>
> = RequiredFlagsParam<CommandT, RequiredParamsT> & OptionalFlagsParam<CommandT, OptionalParamsT>;
