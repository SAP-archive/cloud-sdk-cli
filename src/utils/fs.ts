/* Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. */

import { promisify } from 'util';
import * as fs from 'fs';
import * as rimraf from 'rimraf';

export const access = promisify(fs.access);
export const mkdir = promisify(fs.mkdir);
export const readdir = promisify(fs.readdir);
export const readFile = promisify(fs.readFile);
export const copyFile = promisify(fs.copyFile);
export const writeFile = promisify(fs.writeFile);
export const rm = promisify(rimraf);
