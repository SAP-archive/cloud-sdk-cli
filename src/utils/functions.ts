/*!
 * Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved.
 */

export interface Confirm {
  confirm: (message: string) => Promise<boolean>;
}

function confirm(message: string): Promise<boolean> {
  return Promise.resolve(true);
}

export const testFunctions = {
  confirm
};
