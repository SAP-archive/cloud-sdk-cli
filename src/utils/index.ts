/*!
 * Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved.
 */

export * from './copy-list';
export * from './generate-vdm-util';
export * from './git-ignore';
export * from './jest-config';
export * from './manifest-yaml';
export * from './message-formatter';
export * from './package-json';
export * from './scaffold';
export * from './templates';
export * from './usage-analytics';
export * from './warnings';
