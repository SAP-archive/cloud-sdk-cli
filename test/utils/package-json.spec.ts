/*!
 * Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved.
 */
jest.mock('execa', () => jest.fn().mockResolvedValue('1.0.0'));

import * as execa from 'execa';
import * as fs from 'fs';
import * as path from 'path';
import * as rm from 'rimraf';
import { installDependencies, modifyPackageJson, parsePackageJson } from '../../src/utils';
import { getCleanProjectDir, getTestOutputDir } from '../test-utils';

const testOutputDir = getTestOutputDir(__filename);

describe('Package Json Utils', () => {
  afterAll(() => {
    rm.sync(testOutputDir);
  });

  it('should call `npm install`', () => {
    installDependencies('', true);
    expect(execa).toHaveBeenCalledWith('npm', ['install'], { cwd: '', stdio: 'inherit' });
  });

  it('should parse the package.json', () => {
    const projectDir = getCleanProjectDir(testOutputDir, 'parse-package-json');
    const packageJsonPath = path.resolve(projectDir, 'package.json');
    fs.copyFileSync(path.resolve('test', 'nest', 'package.json'), packageJsonPath);

    expect(Object.keys(parsePackageJson(projectDir)).sort()).toMatchSnapshot();
  });

  it('add scripts, dependencies and test config for existing project', async () => {
    const projectDir = getCleanProjectDir(testOutputDir, 'modify-package-json-existing');
    const packageJsonPath = path.resolve(projectDir, 'package.json');
    fs.copyFileSync(path.resolve('test', 'nest', 'package.json'), packageJsonPath);

    await modifyPackageJson({
      projectDir,
      isScaffold: false,
      frontendScripts: false,
      force: false,
      addCds: false
    });
    expect(parsePackageJson(projectDir)).toMatchSnapshot();
  });

  it('add scripts, dependencies and test config for scaffolded project', async () => {
    const projectDir = getCleanProjectDir(testOutputDir, 'modify-package-json-existing');
    const packageJsonPath = path.resolve(projectDir, 'package.json');
    fs.copyFileSync(path.resolve('test', 'nest', 'package.json'), packageJsonPath);

    await modifyPackageJson({
      projectDir,
      isScaffold: true,
      frontendScripts: true,
      force: false,
      addCds: false
    });
    expect(parsePackageJson(projectDir)).toMatchSnapshot();
  });
});
