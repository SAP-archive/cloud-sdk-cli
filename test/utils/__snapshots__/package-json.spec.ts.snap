// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Package Json Utils add scripts, dependencies and test config for existing project 1`] = `
Object {
  "author": "",
  "dependencies": Object {
    "@nestjs/common": "^6.7.2",
    "@nestjs/core": "^6.7.2",
    "@nestjs/platform-express": "^6.7.2",
    "@sap-cloud-sdk/core": "^undefined",
    "reflect-metadata": "^0.1.13",
    "rimraf": "^3.0.0",
    "rxjs": "^6.5.3",
  },
  "description": "",
  "devDependencies": Object {
    "@nestjs/cli": "^6.9.0",
    "@nestjs/schematics": "^6.7.0",
    "@nestjs/testing": "^6.7.1",
    "@sap-cloud-sdk/cli": "^undefined",
    "@sap-cloud-sdk/test-util": "^undefined",
    "@types/express": "^4.17.1",
    "@types/jest": "^24.0.18",
    "@types/node": "^12.7.5",
    "@types/supertest": "^2.0.8",
    "jest": "^24.9.0",
    "prettier": "^1.18.2",
    "supertest": "^4.0.2",
    "ts-jest": "^24.1.0",
    "ts-loader": "^6.1.1",
    "ts-node": "^8.4.1",
    "tsconfig-paths": "^3.9.0",
    "tslint": "^5.20.0",
    "typescript": "^3.6.3",
  },
  "jest": Object {
    "coverageDirectory": "../coverage",
    "moduleFileExtensions": Array [
      "js",
      "json",
      "ts",
    ],
    "rootDir": "src",
    "testEnvironment": "node",
    "testRegex": ".spec.ts$",
    "transform": Object {
      "^.+\\\\.(t|j)s$": "ts-jest",
    },
  },
  "license": "MIT",
  "name": "nest",
  "scripts": Object {
    "build": "nest build",
    "ci-backend-unit-test": "echo \\"Test your application and write results in a JUnit format to \`s4hana_pipeline/reports/backend-unit/\` and coverage in a cobertura format to \`s4hana_pipeline/reports/coverage/backend-unit/\`\\"",
    "ci-build": "echo \\"Use this to compile or minify your application\\"",
    "ci-integration-test": "echo \\"Test your application and write results in a JUnit format to \`s4hana_pipeline/reports/backend-integration/\` and coverage in a cobertura format to \`s4hana_pipeline/reports/coverage/backend-integration/\`\\"",
    "ci-package": "sap-cloud-sdk package --include=\\"package.json,package-lock.json,index.js,dist/**/*\\"",
    "format": "prettier --write \\"src/**/*.ts\\" \\"test/**/*.ts\\"",
    "lint": "tslint -p tsconfig.json -c tslint.json",
    "prebuild": "rimraf dist",
    "start": "nest start",
    "start:debug": "nest start --debug --watch",
    "start:dev": "nest start --watch",
    "start:prod": "node dist/main",
    "test": "jest",
    "test:cov": "jest --coverage",
    "test:debug": "node --inspect-brk -r tsconfig-paths/register -r ts-node/register node_modules/.bin/jest --runInBand",
    "test:e2e": "jest --config ./test/jest-e2e.json",
    "test:watch": "jest --watch",
  },
  "version": "0.0.1",
}
`;

exports[`Package Json Utils add scripts, dependencies and test config for scaffolded project 1`] = `
Object {
  "author": "",
  "dependencies": Object {
    "@nestjs/common": "^6.7.2",
    "@nestjs/core": "^6.7.2",
    "@nestjs/platform-express": "^6.7.2",
    "@sap-cloud-sdk/core": "^undefined",
    "reflect-metadata": "^0.1.13",
    "rimraf": "^3.0.0",
    "rxjs": "^6.5.3",
  },
  "description": "",
  "devDependencies": Object {
    "@nestjs/cli": "^6.9.0",
    "@nestjs/schematics": "^6.7.0",
    "@nestjs/testing": "^6.7.1",
    "@sap-cloud-sdk/cli": "^undefined",
    "@sap-cloud-sdk/test-util": "^undefined",
    "@types/express": "^4.17.1",
    "@types/jest": "^24.0.18",
    "@types/node": "^12.7.5",
    "@types/supertest": "^2.0.8",
    "jest": "^24.9.0",
    "jest-junit": "^undefined",
    "prettier": "^1.18.2",
    "supertest": "^4.0.2",
    "ts-jest": "^24.1.0",
    "ts-loader": "^6.1.1",
    "ts-node": "^8.4.1",
    "tsconfig-paths": "^3.9.0",
    "tslint": "^5.20.0",
    "typescript": "^3.6.3",
  },
  "jest": Object {
    "collectCoverage": true,
    "coverageDirectory": "../s4hana_pipeline/reports/coverage-reports/backend-unit",
    "coverageReporters": Array [
      "text",
      "cobertura",
    ],
    "moduleFileExtensions": Array [
      "js",
      "json",
      "ts",
    ],
    "reporters": Array [
      "default",
      Array [
        "jest-junit",
        Object {
          "outputDirectory": "./s4hana_pipeline/reports/backend-unit",
          "suiteName": "backend unit tests",
        },
      ],
    ],
    "rootDir": "src",
    "testEnvironment": "node",
    "testRegex": ".spec.ts$",
    "transform": Object {
      "^.+\\\\.(t|j)s$": "ts-jest",
    },
  },
  "license": "MIT",
  "name": "nest",
  "scripts": Object {
    "build": "nest build",
    "ci-backend-unit-test": "jest --ci",
    "ci-build": "npm run build",
    "ci-e2e": "echo \\"Test your application and write results in a JUnit format to \`s4hana_pipeline/reports/e2e/\`\\"",
    "ci-frontend-unit-test": "echo \\"Test your application and write results in a JUnit format to \`s4hana_pipeline/reports/frontend-unit/\` and coverage in a cobertura format to \`s4hana_pipeline/reports/coverage/frontend-unit/\`\\"",
    "ci-integration-test": "jest --ci --config ./test/jest-e2e.json",
    "ci-package": "sap-cloud-sdk package",
    "deploy": "npm run ci-build && npm run ci-package && cf push",
    "format": "prettier --write \\"src/**/*.ts\\" \\"test/**/*.ts\\"",
    "lint": "tslint -p tsconfig.json -c tslint.json",
    "prebuild": "rimraf dist",
    "start": "nest start",
    "start:debug": "nest start --debug --watch",
    "start:dev": "nest start --watch",
    "start:prod": "node dist/main",
    "test": "jest",
    "test:cov": "jest --coverage",
    "test:debug": "node --inspect-brk -r tsconfig-paths/register -r ts-node/register node_modules/.bin/jest --runInBand",
    "test:e2e": "jest --config ./test/jest-e2e.json",
    "test:watch": "jest --watch",
  },
  "version": "0.0.1",
}
`;

exports[`Package Json Utils should parse the package.json 1`] = `
Array [
  "author",
  "dependencies",
  "description",
  "devDependencies",
  "jest",
  "license",
  "name",
  "scripts",
  "version",
]
`;
